feat: display extension icons

vzhukovs opened the pull request:
### What does this PR do?
Current changes proposal adds An ability to display extension icons.

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### Screenshot/screencast of this PR

<img width="1252" alt="Снимок экрана 2022-12-19 в 14 55 13" src="https://user-images.githubusercontent.com/1968177/208430925-c62b0809-092f-42ca-9fff-85ab56fc1375.png">


### What issues does this PR fix or reference?

#393 

### How to test this PR?

Try to install Docker Desktop Extension:

```
docker/disk-usage-extension:0.2.5
docker/logs-explorer-extension:0.2.1
```

vzhukovs commented:
cc @slemeur

slemeur commented:
Thanks @vzhukovs !

It feels to me the that alignement of the extension name is inconsistent. 

![unnamed](https://user-images.githubusercontent.com/1636769/208432550-ed9f0319-68b0-4c3f-ac30-1ef4a60e22cd.png)

On the other side, it feels that the icon of the extensions all have a different size. Is there a way for us to improve this?

vzhukovs commented on packages/main/src/plugin/contribution-manager.ts:
```diff
@@ -78,6 +81,26 @@ export class ContributionManager {
+    if (!fs.existsSync(iconPath)) {
+      throw new Error('Invalid icon path : ' + iconPath);
+    }
+    return new Promise((resolve, reject) => {
```
updated

vzhukovs commented:
>On the other side, it feels that the icon of the extensions all have a different size. Is there a way for us to improve this?

Each icon is scaled to 24x24px size, and read directly from the each extension. Not sure if we can update for example positioning.

>It feels to me the that alignement of the extension name is inconsistent.

Will try to take a look, how we can align items in this list to one base line.

cdrage commented:
off topic, but I think we should do the same for providers

vzhukovs commented:
@benoitf @cdrage I've updated the PR, fixed issue with the icons aligning:
![extension icons](https://user-images.githubusercontent.com/1968177/216961269-23ad5d9c-0f73-4988-87c0-469ab8435c64.gif)

cdrage commented on packages/main/src/plugin/contribution-manager.ts:
```diff
@@ -78,6 +81,20 @@ export class ContributionManager {
     this.apiSender.send('contribution-register', this.contributions);
   }
 
+  async loadBase64Icon(rootDirectory: string, metadata: any): Promise<string> {
```
lint error, no `any` on metadata / something else on it?

vzhukovs replied:
`metadata` is a raw representation of package.json, I think, we can add
`// eslint-disable-next-line @typescript-eslint/no-explicit-any`
as it has been done for `loadMetadata` method, wdyt?

vzhukovs replied:
added a comment to the line

cdrage reviewed (changes requested):
1 small comment, otherwise LGTM. Integration works wonderfully, icons are loaded and changing the size of the window updates it / alignts it correctly.

cdrage commented:
LGTM

benoitf commented on packages/renderer/src/AppNavigation.svelte:
```diff
@@ -49,7 +50,7 @@ export let meta;
 <svelte:window bind:innerWidth="{innerWidth}" />
 <nav
-  class="pf-c-nav z-0 group w-12 hover:w-[180px] md:w-[180px] md:min-w-[180px] shadow flex-col justify-between sm:flex transition-all duration-500 ease-in-out"
+  class="pf-c-nav z-0 group w-14 hover:w-[250px] md:w-[250px] hover:sm:w-[250px] md:min-w-[200px] shadow flex-col justify-between sm:flex transition-all duration-500 ease-in-out overflow-hidden hover:overflow-y-auto"
```
It increases the width of the left navbar no ?

vzhukovs replied:
Width of collapsed left bar was increased to have equal spacing on the icons:

<details>
  <summary>Before update:</summary>

![Monosnap Podman Desktop 2023-02-09 12-42-10](https://user-images.githubusercontent.com/1968177/217793579-df264cd5-c571-4f39-8fde-18e57e806869.jpg)
</details>

<details>
  <summary>After setting width to 3.5rem:</summary>

![Monosnap Podman Desktop 2023-02-09 12-37-22](https://user-images.githubusercontent.com/1968177/217793956-abf1f640-ce92-4c8d-86ea-6544f43594b7.jpg)

</details>

Width of the left navigation bar was increased a bit to not to have issues with displaying extension list:

<details>
  <summary>Before update(default opening state):</summary>

![Monosnap Podman Desktop 2023-02-09 12-48-03](https://user-images.githubusercontent.com/1968177/217794448-1bbeae7b-5613-4c63-88f1-729385946020.jpg)

</details>

<details>
  <summary>After setting new width values(default opening state):</summary>

![Monosnap Podman Desktop 2023-02-09 12-39-55](https://user-images.githubusercontent.com/1968177/217794610-689b6cb1-99f6-4b63-9d03-de4e55623976.jpg)

</details>


<details>
  <summary>Before update(scaled window to 1300px):</summary>

![Monosnap Podman Desktop 2023-02-09 12-46-40](https://user-images.githubusercontent.com/1968177/217795089-aea54357-159e-4661-8529-7f39a263ec80.jpg)

</details>


<details>
  <summary>After update(scaled window to 1300px):</summary>

![Monosnap Podman Desktop 2023-02-09 12-40-24](https://user-images.githubusercontent.com/1968177/217795294-c6feef2a-6fb5-410e-80c6-9bbf67edd30f.png)

</details>

Visually, difference is little, but extension list looks prettier.

lstocchi reviewed (commented):
Icons LGTM
On Windows the row doesn't take the full width. It would be better to have it
![highlight_too_short](https://user-images.githubusercontent.com/49404737/217039256-0a922fc3-a8bc-42b4-a9ba-62521ee331f6.gif)

vzhukovs commented on packages/main/src/plugin/contribution-manager.ts:
```diff
@@ -78,6 +81,21 @@ export class ContributionManager {
+  async loadBase64Icon(rootDirectory: string, metadata: any): Promise<string> {
+    if (!metadata.icon) {
+      //return an empty svg icon <svg xmlns="http://www.w3.org/2000/svg" width="1" height="1"/>
+      return 'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxIiBoZWlnaHQ9IjEiLz4=';
```
updated

vzhukovs replied:
Width of collapsed left bar was increased to have equal spacing on the icons:

<details>
  <summary>Before update:</summary>

![Monosnap Podman Desktop 2023-02-09 12-42-10](https://user-images.githubusercontent.com/1968177/217793579-df264cd5-c571-4f39-8fde-18e57e806869.jpg)
</details>

<details>
  <summary>After setting width to 3.5rem:</summary>

![Monosnap Podman Desktop 2023-02-09 12-37-22](https://user-images.githubusercontent.com/1968177/217793956-abf1f640-ce92-4c8d-86ea-6544f43594b7.jpg)

</details>

Width of the left navigation bar was increased a bit to not to have issues with displaying extension list:

<details>
  <summary>Before update(default opening state):</summary>

![Monosnap Podman Desktop 2023-02-09 12-48-03](https://user-images.githubusercontent.com/1968177/217794448-1bbeae7b-5613-4c63-88f1-729385946020.jpg)

</details>

<details>
  <summary>After setting new width values(default opening state):</summary>

![Monosnap Podman Desktop 2023-02-09 12-39-55](https://user-images.githubusercontent.com/1968177/217794610-689b6cb1-99f6-4b63-9d03-de4e55623976.jpg)

</details>


<details>
  <summary>Before update(scaled window to 1300px):</summary>

![Monosnap Podman Desktop 2023-02-09 12-46-40](https://user-images.githubusercontent.com/1968177/217795089-aea54357-159e-4661-8529-7f39a263ec80.jpg)

</details>


<details>
  <summary>After update(scaled window to 1300px):</summary>

![Monosnap Podman Desktop 2023-02-09 12-40-24](https://user-images.githubusercontent.com/1968177/217795294-c6feef2a-6fb5-410e-80c6-9bbf67edd30f.png)

</details>

Visually, difference is little, but extension list looks prettier.

vzhukovs commented:
@lstocchi I've fixed the problem with displaying items on windows, now it looks better:

![Podman+Desktop+2023-02-09+13 32 21](https://user-images.githubusercontent.com/1968177/217795982-3229d7a9-3373-43f0-a008-5a0909fcfd8a.png)

lstocchi reviewed (approved):
It looks fine. Thanks!! 🚀 

![full_width_icon](https://user-images.githubusercontent.com/49404737/217865357-e7ecdfde-2b15-403d-88f9-a83ef99f4d32.gif)

slemeur commented:
Good work @vzhukovs ! LGTM
